packages/core/__tests__/cli/custom-extensions.test.ts

```diff
@@ -135,6 +135,15 @@ describe('CLI: custom extensions', () => {
         'Greeting.gts': stripIndent`
           <template>Hello!</template>
         `,
+        're-export.gts': stripIndent`
+          export { default as Greeting } from './Greeting.gts';
+        `,
+        'vanilla.ts': 'export const two = 2;',
+        'barrel.ts': stripIndent`
+          export { default as Greeting } from './Greeting.gts';
+          export { Greeting as Greeting2 } from './re-export.gts';
+          export { two } from './vanilla.ts';
```

Comment:
the declarations for this are, at present, emitted as `./vanilla.ts`, which is wrong.

To fix, either needs to happen:
- remove the `.ts` extension so `vanilla.d.ts` matches with the emitted corresponding `vanilla.js`
- or emit a goofy `vanilla.ts.d.ts` file, along the same lines as what I suggested originally with emitting `Greeting.gts.d.ts`